reproc/src/reproc.c

```diff
@@ -651,6 +651,19 @@ int reproc_stop(reproc_t *process, reproc_stop_actions stop)
   return r;
 }
 
+int reproc_pid(reproc_t *process)
+{
+  ASSERT_EINVAL(process);
+  ASSERT_EINVAL(process->status != STATUS_IN_CHILD);
+  ASSERT_EINVAL(process->status != STATUS_NOT_STARTED);
+  
+  if (process->status == STATUS_IN_PROGRESS) {
+    return process->handle;
+  } else {
+    return -1;
+  }
```

Comment:
`return process->handle` is OK here, we already check all the preconditions with `ASSERT_EINVAL` so we know its a valid value when we get here so no need to do error handling. 

I think we also want to return the pid even if the process has already stopped. We would need to be careful to mention that the pid might not be usable anymore if `reproc_wait` has been called on it successfully but the check on whether status is `STATUS_IN_PROGRESS` shouldn't be necessary.

I would add this to the function documentation:

> Note that if `reproc_wait` has been called successfully on this process already, the operating system will have cleaned up the resources allocated to the process and the operating system is free to reuse the same pid for another process. Generally, only pass the result of this function to system calls that need a pid if `reproc_wait` hasn't been called successfully on the process yet.